tests/oidc-provider.test.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const { test } = require('node:test')
+const assert = require('node:assert')
+const http = require('node:http')
+const { request, Agent } = require('undici')
+const { createDecoder, createVerifier } = require('fast-jwt')
+const buildGetJwks = require('get-jwks')
+const { testProvider } = require('./helper.js')
+const { createOidcInterceptor } = require('../')
+
+test('interceptor works with oidc-provider', async (t) => {
+  const jwks = buildGetJwks({
+    jwksPath: '/jwks',
+  })
+  const provider = await testProvider(test, { port: 3000 });
+  const idp = `http://localhost:${provider.address().port}/`
+  const idpTokenUrl = `${idp}token`
+
+  const key = await jwks.getPublicKey({ domain: idp, kid: 'r1LkbBo3925Rb2ZFFrKyU3MVex9T2817Kx0vbi6i_Kc' })
+
+  const mainServer = http.createServer((req, res) => {
+    assert.ok(req.headers.authorization.length > 'Bearer '.length)
+    const verifySync = createVerifier({ key })
+    const decoded = verifySync(req.headers.authorization.slice('Bearer '.length))
+    assert.strictEqual(decoded.iss, `http://localhost:${provider.address().port}`)
+    res.writeHead(200)
+    res.end()
+  })
```

Comment:
```suggestion
  const getKey = ({ header }) => jwks.getPublicKey({ domain: idp, kid: header.kid, alg: header.alg })

  const mainServer = http.createServer(async (req, res) => {
    assert.ok(req.headers.authorization.length > 'Bearer '.length)
    const verifyAsync = createVerifier({ key: getKey })
    const decoded = await verifyAsync(req.headers.authorization.slice('Bearer '.length))
    assert.strictEqual(decoded.iss, `http://localhost:${provider.address().port}`)
    res.writeHead(200)
    res.end()
  })
```
You can use here `verifyAsync` to avoid pinning down to this specific `kid`. 
This way if you later change your jwks set, this test will still be passing.